File: shared/types/location.ts
// Location-related TypeScript type definitions

export interface Location {
  id: string;
  latitude: number;
  longitude: number;
  name?: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export interface LocationWithPhotos extends Location {
  photo_count: number;
}

export interface LocationFilters {
  latitude: number;
  longitude: number;
  radius?: number;
}

export interface Coordinates {
  latitude: number;
  longitude: number;
}

File: server/app/models/location.py
from app import db
from datetime import datetime
import uuid


class Location(db.Model):
    __tablename__ = "locations"

    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    latitude = db.Column(db.Float, nullable=False)
    longitude = db.Column(db.Float, nullable=False)
    name = db.Column(db.String(255))
    description = db.Column(db.Text)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(
        db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow
    )

    def to_dict(self):
        return {
            "id": self.id,
            "latitude": self.latitude,
            "longitude": self.longitude,
            "name": self.name,
            "description": self.description,
            "created_at": self.created_at.isoformat() if self.created_at else None,
            "updated_at": self.updated_at.isoformat() if self.updated_at else None,
        }

    def __repr__(self):
        return f"<Location {self.id}: {self.latitude}, {self.longitude}>"
